### DISCUSSION TITLE
Extract Resources from PE?

### DISCUSSION THREAD

**Discussion by acook** [hasherezade/pe-bear#40] [Q&A] [answered]
Is it possible to export a resource as a file from a PE? 

For example, to extract an icon resource as a valid `.ico` file?

**Comment by hasherezade** [comment, marked as answer]
hi @acook !
Yes, it is possible to export resources:

![export_res](https://github.com/hasherezade/pe-bear/assets/3115348/38f66808-0629-4498-b6e9-e1871875ee35)

However, currently PE-bear export resources only in their raw form, it doesn't append the header to the .ico files during their export. I may implement it as an additional option in the future. For now I would recommend you to use ResourceHacker:
https://angusj.com/resourcehacker/#download

**Reply by acook** [reply]
Ah I was secretly hoping for the headers because otherwise it would be relatively easy to use the offset to extract the data.

I'm on Linux and I haven't tried Resource Hacker in WINE yet but I may do that. Thanks!

**Reply by hasherezade** [reply]
I will take a note and add the option in the next release
